node-src/lib/checkStorybookBaseDir.ts

```diff
@@ -0,0 +1,44 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import { invalidStorybookBaseDir } from '../ui/messages/errors/invalidStorybookBaseDir';
+import { Context, Stats } from '../types';
+import pLimit from 'p-limit';
+import { exitCodes, setExitCode } from './setExitCode';
+
+export async function checkStorybookBaseDir(ctx: Context, stats: Stats) {
+  const { storybookBaseDir } = ctx.options;
+  ctx.log.debug('Storybook base directory:', storybookBaseDir);
+
+  // Find all js(x)/ts(x) files in stats that are not in node_modules
+  const sourceModuleFiles = stats.modules.filter(
+    (module: any) => !module.name.includes('node_modules') && /\.(js|jsx|ts|tsx)$/.test(module.name)
+  );
+
+  // GitHub Actions seems to have a default ulimit of 1024, so we limit concurrency to stay under
+  const limitConcurrency = pLimit(1000);
+
+  // Check if any of the source module files exist in the storybookBaseDir
+  try {
+    await Promise.any(
+      sourceModuleFiles.map((file) => {
+        return limitConcurrency(() => {
+          const absolutePath = path.join(storybookBaseDir || '', file.name);
+          return new Promise((resolve, reject) =>
+            fs.access(absolutePath, (err) => {
+              if (err) {
+                reject();
+              } else {
+                ctx.log.debug('Found:', absolutePath);
+                resolve(true);
+              }
```

Comment:
Can we log the path we **did not** find instead of the ones that we did? It will make it easier to debug and keep the logs cleaner? This could end up being a lot of files.

Reply:
Hmm, you actually have me thinking that either scenario could end up being a lot of files and, thus, a lot of debug prints in the logs. Even if the very first file check succeeds and the promise resolves, the other file checks will still occur. Logging every one of these could be very noisy. On the other hand, the same number of checks could happen in the scenario where none of them succeed. So that could also be noisy.

If we succeed on a check, we really just need that one path logged. Maybe we could bubble that back up to the promise and log it there? If we fail on all checks, would we want a log for every path checked or just one as an example?

Reply:
So, to answer your question, yes we could log files that were **not** found instead of those that were. But that could still be very noisy. I suppose that would be better since we are only potentially logging a lot of files in the unhappy path of an invalid `storybookBaseDir`. That seems like a good improvement.